### PR TITLE
Group improvements

### LINKED ISSUE #741: Prevent users from joining a group if it's already over
Or at least give them a confirmation dialog.

### PR DESCRIPTION
**Pull request by DannySS123**
Users now can't join a group if it's already over, and instead of the join button there is a text saying that this group has expired.
 The "Kilépés" button's text color changed to black in light mode and to white in dark mode in order to match the others and for contrast improvements.
 Closes #741

### CODE REVIEW

**Review by triszt4n — CHANGES_REQUESTED** [review]
Nice. See request below. I'm thinking that maybe we should also inform the user if the group has already started. Maybe an "In session" label after the date in green font color (or after the Join button, but that wouldn't be very aesthetic).

**Comment by Tschonti** [comment]
If you open a group page and then the group expires but you don't refresh, you can still join the group. This isn't big issue, but maybe there could be a check on the server side to make sure it's impossible to join an expired group, especially now that joining a group will send and email.

File: src/components/groups/group.middlewares.ts [hunk 1]
```diff
@@ -23,6 +23,8 @@ export const joinGroup = asyncWrapper(async (req: Request, res: Response, next:
     sendMessage(res, 'Már tagja vagy ennek a csoportnak!')
   } else if ((group.users?.length || 0) >= group.maxAttendees) {
     sendMessage(res, 'Ez a csoport már tele van!')
+  } else if (group.endDate < new Date()) {
+    sendMessage(res, 'Ez a csoport már lejárt!')
```
Comment (OmTheTurtle) [inline src/components/groups/group.middlewares.ts:27]: I'd change the phrasing a little bit, what do you think of 'Ez a csoport már véget ért!'?

File: views/group/show.pug [hunk 2]
```diff
@@ -5,13 +5,19 @@ include ../modalTemplate.pug
...
   - const [endDate, endTime] = format(group.endDate, DATE_FORMAT).split(' ')
+  - const groupEnded = group.endDate < new Date()
+  - const groupStarted = group.startDate < new Date()
   div(class='flex flex-row items-center mb-8 mr-4 space-x-4')
     h1(class='truncate title')= group.name
-    if !joined && !group.doNotDisturb
+    if groupStarted && !groupEnded
+      p(class="text-2xl text-green-700") Már elkezdődött.
+    else if groupEnded
+      p(class="text-2xl text-red-700") Ez a csoport lejárt.
```
Comment (OmTheTurtle) [inline views/group/show.pug:15]: Same phrasing as above.

**Review by OmTheTurtle — APPROVED** [review]

**Pull request merged**